.github/workflows/check.yml

```diff
@@ -10,4 +10,11 @@ jobs:
         with:
           ruby-version: '3.2'
           bundler-cache: true # runs 'bundle install' and caches installed gems automatically
-      - run: bundle exec rake typecheck
+      - name: Check for stale signature files
+        run: bundle exec rake rbs:stale
+      - name: Check for missing signature files
+        run: bundle exec rake rbs:missing
```

Comment:
From: https://github.com/DataDog/dd-trace-rb/pull/2859#discussion_r1198833448

> We can tweak the check inside the GitHub action using ${{github.actor}} information. That way we do not bother the community with adding types. Not suggesting doing on this PR

> +1 if this can be gated only to the "usual suspects" I have no concerns at all with the approach and think it's a great idea.

I am concerned that this conditional skip would lead to an immediate failure on the merge commit of a contributor PR.

I don't think it's too much to ask for rake `rbs:stale` and `rake rbs:prototype` to be called (which is now suggested in the failure), and the user adding an `ignore` to `Steepfile` as suggested by `rake steep:check` if types are too gnarly to fix e.g prototyping fails to generate a passing skeleton.

Note that the latter case may happen already if someone changes an existing, previously type checked file and type checks suddenly don't pass.

Reply:
I believe what @ivoanjo was concerned about was not with telling users to run those commands `clean`, and `prototype`  but more with the idea of forcing users, not maintainers (ourselves), to make sure to type (make sure no `untyped` is present in the files they have modified with the PR. 

The main idea comes for gradually adding types in the `sig` files rather than just having `untyped`